.github/workflows/ci_build_ios.yml

```diff
@@ -0,0 +1,63 @@
+name: Build iOS library
+
+on:
+  workflow_call:
+    outputs:
+      artifact-dir-base:
+        description: The base name of artifact directory (without the target suffix)
+        value: ${{ jobs.build-ios-libs.outputs.artifact-dir-base }}
+
+jobs:
+  build-ios-matrix:
+    runs-on: ubuntu-20.04
+    timeout-minutes: 5
+    outputs:
+      matrix: ${{ steps.set-matrix.outputs.matrix }}
+    steps:
+      - uses: actions/checkout@ec3a7ce113134d7a93b817d10a8272cb61118579 # v2.4.0
+      - uses: xom9ikk/dotenv@d3ff95524814ceac377510f30f4af6296ea612c1 # v1.0.2
+      - id: set-matrix
+        run: |
+          TARGETS=$(target_array=($IOS_TARGETS); jq --compact-output --null-input '$ARGS.positional' --args "${target_array[@]}")
+          echo "::set-output name=matrix::$(echo {\"target\":$TARGETS})"
+
+  build-ios-libs:
+    needs: build-ios-matrix
+    runs-on: macos-11
+    timeout-minutes: 35
+    strategy:
+      fail-fast: false
```

Comment:
We might want to fail fast here. If one of the build for a target fail the whole ci will fail. I'm not sure trying to build the other one will give us useful information.

Reply:
in the past it was very useful for debugging. For example when we tested the android target we saw that only one architecture failed and not all of them + architectures that build successfully benefit from a cache hit of the artifacts in the next run. but i can also remove it

Reply:
but that was with testing here we only build. if gain from cache we keep it

Reply:
done